packages/core/src/checkout/checkout-action-creator.ts

```diff
@@ -31,11 +31,12 @@ export default class CheckoutActionCreator {
                 }),
                 this._formFieldsActionCreator.loadFormFields({ ...options, useCache: true }),
                 defer(() =>
-                    this._checkoutRequestSender
-                        .loadCheckout(id, options)
-                        .then(({ body }) =>
-                            createAction(CheckoutActionType.LoadCheckoutSucceeded, body),
-                        ),
+                    this._checkoutRequestSender.loadCheckout(id, options).then(({ body }) => {
+                        return createAction(
+                            CheckoutActionType.LoadCheckoutSucceeded,
+                            this._transformCheckout(body),
```

Comment:
should we also transform in `loadDefaultCheckout()`? I don't know when this is called instead of `loadCheckout()` though.

Reply:
Yes, this transform function should be added to `loadDefaultCheckout` as well. Also maybe need to rename `_transformCheckout` to something like `_transformCustomerAddresses` 🤔

Reply:
👍

Reply:
Also can you please write a unit test for this case.

Reply:
I think the unit test is covered? so I changed the mock, because https://github.com/bigcommerce/checkout-sdk-js/blob/b445659272efda3f381f1ed7332d8adef5da566d/packages/core/src/checkout/checkout-action-creator.spec.ts#L85, after called the actionCreator.loadCheckout, it checked the response with expected one. so it should be false now.

Reply:
I will say we should have different test cases for logged in customer vs not logged in customer.

Reply:
not logged in shopper doesn't have addresses in `checkout.customer`. So they don't even have address selection. so our change doesn't affect that flow.